### PR TITLE
Change the way to get subset data in datumaro adapter

### PR DESCRIPTION
**Pull request by jaegukhyun**
### Summary
Current datumaro adapater get any dataset as validation dataset if "val" is in subset name.
Therefore if a dataset has subsets such as ["train", "trainval", "val"] the validation dataset for this dataset would be "trainval".
This PR fixes this issue by using get_close_matches function in difflib.

### How to test
Train with VOC dataset.

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

File: otx/core/data/adapter/base_dataset_adapter.py [hunk 1]
```diff
@@ -162,13 +163,15 @@ def get_label_schema(self) -> LabelSchemaEntity:
...
-                    return v
-                if subset == "test" and "val" in k:
-                    v = v.as_dataset()
-                    return v
+            subsets = list(dataset.subsets().keys())
+
+            for s in [subset, "default"]:
+                if subset == "test":
+                    s = "val"
+                exact_subset = get_close_matches(s, subsets)
```
Comment (supersoob) [inline otx/core/data/adapter/base_dataset_adapter.py:171]: Does it find a closest key name in dataset_subsets with subset from argument?

Comment (jaegukhyun) [reply otx/core/data/adapter/base_dataset_adapter.py:171]: Yes it does

**Review by supersoob — DISMISSED** [review]

**Review by eunwoosh — COMMENTED** [review]
LGTM but could you fix unit test?

**Review by wonjuleee — DISMISSED** [review]

File: otx/core/data/adapter/base_dataset_adapter.py [hunk 2]
```diff
@@ -162,13 +163,23 @@ def get_label_schema(self) -> LabelSchemaEntity:
...
-                if subset == "test" and "val" in k:
-                    v = v.as_dataset()
-                    return v
+            subsets = list(dataset.subsets().keys())
+
+            for s in [subset, "default"]:
+                if subset == "val" and s != "default":
+                    s = "valid"
+                exact_subset = get_close_matches(s, subsets)
+                print(subset, s, subsets, exact_subset)
```
Comment (wonjuleee) [inline otx/core/data/adapter/base_dataset_adapter.py]: please remove this.

Comment (jaegukhyun) [reply otx/core/data/adapter/base_dataset_adapter.py]: Done!

**Review by wonjuleee — APPROVED** [review]

**Review by eunwoosh — APPROVED** [review]

**Review by cih9088 — APPROVED** [review]

**Pull request merged**